### ISSUE TITLE
intermittent failures in `test_xor_training_correctness[LLVMRun-50-single-adam-False]`

### ISSUE THREAD

**Issue by jvesely** [PrincetonUniversity/PsyNeuLink#1545]
see for example:
https://ci.appveyor.com/project/jvesely/psyneulink-coveralls/builds/31663125
 a repeated build of the same code succeeds:
https://ci.appveyor.com/project/jvesely/psyneulink-coveralls/builds/31663917

**Comment by jvesely** [comment]
cc @SamKG

**Comment by SamKG** [comment]
Will take a look at this - I suspect this bug was introduced when I had removed ref passing for the weights

**Comment by jvesely** [comment]
I checked the test, and it doesn't hit the path fixed in #1610.
Also, the results look wrong even for Python mode, printing them shows:
```
tests/composition/test_autodiffcomposition.py::TestTrainingCorrectness::test_xor_training_correctness[Python-50-single-adam-True] [[array([0.31200036])], [array([0.59406178])], [array([0.60417587])], [array([0.52347365])]]
PASSED
tests/composition/test_autodiffcomposition.py::TestTrainingCorrectness::test_xor_training_correctness[Python-50-single-adam-False] [[array([0.399898])], [array([0.58961467])], [array([0.54293491])], [array([0.51298241])]]
PASSED
tests/composition/test_autodiffcomposition.py::TestTrainingCorrectness::test_xor_training_correctness[LLVMRun-50-single-adam-True] [[[0.31202413946754987]], [[0.5940729063910729]], [[0.6041634977020353]], [[0.5234295358656476]]]
PASSED
tests/composition/test_autodiffcomposition.py::TestTrainingCorrectness::test_xor_training_correctness[LLVMRun-50-single-adam-False] [[[0.43985781659105583]], [[0.5190865881628404]], [[0.5582109612930329]], [[0.5144072619250126]]]
PASSED
```

while the `True` results look similar between Python and LLVM. The `False` results differ widely.

Not that the 4th result rounds to `1` even though it's expected to be `0`.
The tests pass because the results comparison loop checks only the first result and ignores the rest.

**Comment by SamKG** [comment]
I'll take another pass on verifying correctness; The gap between Python and LLVM looks way larger than previously seen, so it's possible there is some discrepancy between the logic in both.

**Comment by kmantel** [comment]
I ran into this on the 100 variant, see 
[Jenkins 3.6 Build #4101](https://github.com/PrincetonUniversity/PsyNeuLink/files/4699463/jenkins-py36-4100.txt)
 and the successful actions build https://github.com/PrincetonUniversity/PsyNeuLink/actions/runs/118555296

Should we xfail this test for now?

**Comment by jvesely** [comment]
> I ran into this on the 100 variant, see
> [Jenkins 3.6 Build #4101](https://github.com/PrincetonUniversity/PsyNeuLink/files/4699463/jenkins-py36-4100.txt)
> and the successful actions build https://github.com/PrincetonUniversity/PsyNeuLink/actions/runs/118555296
> 
> Should we xfail this test for now?

xfail would expect failure and complain if it passes. I think it's good that it occasionally pops out, stays on the radar

**Comment by jvesely** [comment]
Fixed in #1648
